### PR TITLE
Log syncclient status code in statsd rather than logging the exception in Sentry.

### PR DESCRIPTION
**Pull request by Natim**
```
"GET   /v1/buckets/syncto/collections/meta/records" ? (? ms) 401 Client Error: Unauthorized for url: https://token.services.mozilla.com/1.0/sync/1.5?duration=300 lang=en-US,en;q=0.5; exception=Traceback (most recent call last):
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/router.py", line 163, in handle_request
    response = view_callable(context, request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/config/views.py", line 596, in __call__
    return view(context, request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/config/views.py", line 329, in attr_view
    return view(context, request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/config/views.py", line 305, in predicate_wrapper
    return view(context, request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/config/views.py", line 355, in rendered_view
    result = view(context, request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/pyramid/config/views.py", line 501, in _requestonly_view
    response = view(request)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/cornice/service.py", line 573, in wrapper
    response = view_(request)
  File "/home/ubuntu/syncto/syncto/views/collection.py", line 21, in collection_get
    sync_client = build_sync_client(request)
  File "/home/ubuntu/syncto/syncto/authentication.py", line 65, in build_sync_client
    credentials = tokenserver.get_hawk_credentials(duration=ttl)
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/syncclient/client.py", line 68, in get_hawk_credentials
    raw_resp.raise_for_status()
  File "/home/ubuntu/venvs/syncto/local/lib/python2.7/site-packages/requests/models.py", line 837, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 401 Client Error: Unauthorized for url: https://token.services.mozilla.com/1.0/sync/1.5?duration=300; uid=None; errno=None; agent=Mozilla/5.0 (Mobile; rv:43.0) Gecko/20100101 Firefox/43.0; authn_type=None
```

### CODE REVIEW

**Comment by Natim** [comment]
- [x] Add test to make sure statsd is called.

**Comment by Natim** [comment]
Updated with @ametaireau insights on IRC.

File: syncto/views/collection.py [hunk 1]
```diff
@@ -61,6 +61,10 @@ def collection_get(request):
     records = sync_client.get_records(collection_name, full=True,
                                       headers=headers, **params)
 
+    statsd = request.registry.statsd
```
Comment (almet) [inline syncto/views/collection.py:64]: should we provide an helper for this? `statsd(request, "count", "syncclient.status_code.200")`

**Comment by almet** [comment]
r+wc

File: syncto/views/collection.py (hunk 1 above)
Comment (Natim) [inline syncto/views/collection.py:64]: Good idea, I have created https://github.com/mozilla-services/cliquet/issues/480 about it.

**Pull request merged**